### ISSUE TITLE
Supporting slicing and indexing semantics

### ISSUE THREAD

**Issue by philippjfr** [CubeBrowser/cube-explorer#12]
In HoloViews most types can be sliced by their values using the `__getitem__` and select methods. This makes it very easy to select subsets of the data held in a HoloViews object, particularly when that object has been expanded into a nested datastructure. While this is very convenient we wouldn't want to reimplement something that Iris already supports well or provide an interface that clashes with the existing Iris semantics. Therefore we should discuss how we can provide an API on HoloCube based Elements that wraps the Iris slicing and indexing implementation in a meaningful way.

**Comment by philippjfr** [comment]
Note that implementing `__getitem__` and `select` will be required if we want to support dynamic groupby throughout HoloViews. So I think this has greater priority than the `wishlist` tag indicates. It shouldn't be particularly difficult to translate HoloViews slicing semantics into Iris Constraint objects to support slicing.

**Comment by marqh** [comment]
closed by #53

**Comment by marqh** [comment]
change this line in the docs:

```
temp_curve = surface_temperature.to.curve('time', dynamic=True)[0, 10]
```
